test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -1017,3 +1021,94 @@ func TestCRDDeletionCascading(t *testing.T) {
 		t.Fatalf("failed waiting for dependent %q (owned by %q) to be deleted", dependent.GetName(), owner.GetName())
 	}
 }
+
+func TestReplicaSetsAppsV1DefaultGCPolicy(t *testing.T) {
```

Comment:
Is there any reason we shouldn't put this one in the `replicaset` package too?

Reply:
@enisoc Do we need to add similar tests for `Deployment`, `DaemonSet` and `StatefulSet` as well?